src/Illuminate/Database/Eloquent/Relations/Pivot.php

```diff
@@ -92,6 +105,10 @@ public static function fromRawAttributes(Model $parent, $attributes, $table, $ex
      */
     protected function setKeysForSaveQuery(Builder $query)
     {
+        if ($this->instantiatedAsModel) {
```

Comment:
The problem stems from Pivot breaking substitution with Model... I'll propose some rearranging of the Pivot class to prevent the need for this.